### PR TITLE
[Feature:System] Backend for Community Events banners

### PR DESCRIPTION
**Pull request by 04vmatsibekker**
### Please check if the PR fulfills these requirements:

* [ ] Tests for the changes have been added/updated (if possible)
* [ ] Documentation has been updated/added if relevant
* [ ] Screenshots are attached to Github PR if visual/UI changes were made

### What is the current behavior?

Can't displays advertisements for clubs
### What is the new behavior?

A faculty member, you can upload at most 2 banners, one will be a smaller banner that is shown on the screen, and one is a bigger (optional) banner that gets displayed when clicking the smaller banner, you can also have the banner go to a link when pressed. 

This uploads files to the community_events folder. 

image

### CODE REVIEW

**Review by DarthNyan — CHANGES_REQUESTED** [review]
There were some areas, especially in PHP code, where I had trouble following what you were doing and why it was necessary. Adding code comments throughout would help. A comment that explains the banner file structure would also be helpful for maintainers, as would comments describing the data fields in the BannerImage class.

I also noticed some changes to the poll code that seem like they belong to a separate PR.

Other than that, things look good to me.

File: site/app/templates/polls/PollForm.twig
Comment (04vmatsibekker) [reply site/app/templates/polls/PollForm.twig:5]: It makes my cypress test pass

File: site/app/templates/banner/UploadBannerPopup.twig
Comment (04vmatsibekker) [reply site/app/templates/banner/UploadBannerPopup.twig:70]: It causes complications to move it, this can be a separate PR

File: site/app/repositories/banner/BannerImageRepository.php
Comment (williamjallen) [reply site/app/repositories/banner/BannerImageRepository.php:18]: Regular SQL queries go in `DatabaseQueries.php`, but this is a Doctrine DQL query.  DQL queries belong in "Repositories" associated with a Doctrine entity.

**Comment by williamjallen** [comment]
It would be helpful to fill out the PR description with details about why this change is made, and the future roadmap for this feature.  If another developer takes on the frontend later on, having information to refer to will be immensely helpful.

File: site/app/controllers/banner/BannerController.php
Comment (04vmatsibekker) [reply site/app/controllers/banner/BannerController.php:69]: Cuz you can only have 1 banner, and 1 bigger banner

File: site/app/controllers/banner/BannerController.php
Comment (cjreed121) [inline site/app/controllers/banner/BannerController.php:114]: Isn't `count_item` restricted to be 1? Can we remove this loop?

**Review by cjreed121 — CHANGES_REQUESTED** [review]

File: site/public/js/drag-and-drop.js
Comment (williamjallen) [inline site/public/js/drag-and-drop.js:212]: Did you copy this from somewhere?

File: site/public/js/server.js
Comment (williamjallen) [inline site/public/js/server.js:14]: This should be moved to the existing `exported` section at the top

**Review by williamjallen — CHANGES_REQUESTED** [review]
Other than these comments, I think it's good enough to be considered done as a work-in-progress at this point.  There are a bunch of small stylistic things left, but they probably aren't worth blocking this further.

**Review by DarthNyan — APPROVED** [review]
Looks good to me.

**Comment by williamjallen** [comment]
The current PR title does not adequately describe what this PR does.  Please change the title to something that summarizes this PR so it's clear in the release notes.

The changes seem reasonable to me, and I'll approve once the title is changed.

**Comment by 04vmatsibekker** [comment]
> The current PR title does not adequately describe what this PR does. Please change the title to something that summarizes this PR so it's clear in the release notes.
> 
> The changes seem reasonable to me, and I'll approve once the title is changed.

Done

**Review by williamjallen — APPROVED** [review]
Seems reasonable to me as a first step towards the final feature implementation.

**Review by cjreed121 — APPROVED** [review]
Changes look good for the backend of this feature

**Review by bmcutler — APPROVED** [review]
I tweaked text and switched access to superusers only to prevent confusing faculty with unfinished feature.

Ready to merge assuming tests pass...

**Pull request merged**
